src/features/account/components/modals/AddToUserCircleModal.tsx

```diff
@@ -26,50 +28,75 @@ export const AddToUserCircleModal: FunctionComponent<
 
-  const {
-    isLoading,
-    isVerified,
-    payload,
-    submitTypedSignRequest,
-    signatureError,
-  } = useTypedSignTrustCredential();
+  const [isLoading, setIsLoading] = useState(false);
+  const { issuerAddress, signMessage, signError, accountVCBuilder } =
+    useVerifiableCredential();
 
-  const { showErrorMsg, showSuccessMsg } = useToastMsg();
+  const { showSuccessMsg, showErrorMsg } = useToastMsg();
 
   const shortSubAddress = useMemo(
     () => trimAddress(subjectAddress),
     [subjectAddress],
   );
 
-  useEffect(() => {
-    if (signatureError) {
-      if (signatureError.type === SignatureErrorTypes.Error) {
-        showErrorMsg({
-          title: t`Failed to verify signature`,
-          description: signatureError.message,
-        });
-      } else {
+  const onSignButtonClick = useCallback(() => {
+    if (!issuerAddress) {
+      return;
+    }
+    if (!isAddress(subjectAddress)) {
+      return;
+    }
+    setIsLoading(true);
+
+    const VC = accountVCBuilder.buildAccountTrust(
+      issuerAddress,
+      subjectAddress,
+    );
+
+    signMessage(VC)
+      .then((signature) => {
+        if (signature) {
+          const assertion = accountVCBuilder.getSignedAssertion(VC, signature);
+          console.log('assertion', assertion);
+
+          showSuccessMsg({
+            title: t`Added to your trust circle`,
+            description: t`${shortSubAddress} has been added to your trust circle`,
+          });
+          dispatch(addUserToUserCircle(subjectAddress));
+          dispatch(setAddToUserModalOpen(false));
+        } else {
+          showErrorMsg({
+            title: t`Error`,
+            description: t`The signature is empty`,
+          });
+        }
+      })
+      .catch((error) => {
+        console.error(error);
         showErrorMsg({
-          title: t`Invalid Signature`,
-          description: t`Your signature is invalid`,
+          title: t`Error`,
+          description: t`Unknown error`,
         });
-      }
-    }
-  }, [signatureError, showErrorMsg]);
+      });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [issuerAddress, subjectAddress]);
 
   useEffect(() => {
-    // confirm signature has been verified
-    if (isVerified) {
-      console.log(`TrustCredential payload is`, payload);
-      showSuccessMsg({
-        title: t`Added to your trust circle`,
-        description: t`${shortSubAddress} has been added to your trust circle`,
+    if (signError?.type === VCSignErrorType.SignError) {
+      showErrorMsg({
+        title: t`Error`,
+        description: t`The signature could not be created`,
+      });
+      setIsLoading(false);
+    } else if (signError?.type === VCSignErrorType.VerifyFailed) {
+      showErrorMsg({
+        title: t`Error`,
+        description: t`The signature verification failed`,
       });
-      dispatch(addUserToUserCircle(subjectAddress));
-      dispatch(setAddToUserModalOpen(false));
+      setIsLoading(false);
```

Comment:
```suggestion

```

Reply:
this move to signMessage

Reply:
Latest change have remove all setIsLoading(false) in useEffect, and handle at `onSignButtonClick`